sqlalchemy_exasol/base.py

```diff
@@ -661,63 +667,85 @@ def get_view_definition_sql(self, connection, view_name, schema=None, **kw):
         else:
             return None
 
+    def get_column_sql_query_str(self):
+        return "SELECT " \
+               "column_name, " \
+               "column_type, " \
+               "column_maxsize, " \
+               "column_num_prec, " \
+               "column_num_scale, " \
+               "column_is_nullable, " \
+               "column_default, " \
+               "column_identity, " \
+               "column_is_distribution_key, " \
+               "column_table " \
+               "FROM sys.exa_all_columns " \
+               "WHERE " \
+               "column_object_type IN ('TABLE', 'VIEW') AND " \
+               "column_schema = {schema} AND " \
+               "column_table = {table} " \
+               "ORDER BY column_ordinal_position"
+
     @reflection.cache
-    def _get_all_columns_odbc(self, connection, odbc_connection, schema, **kw):
+    def _get_all_columns_odbc(self, connection, odbc_connection, table_name, schema, **kw):
         schema = self._get_schema_for_input_or_current(connection, schema)
         # We need to check if the schema is valid
-        tables = self._get_tables_for_schema_odbc(connection, odbc_connection, schema=schema)
+        tables = self._get_tables_for_schema_odbc(connection, odbc_connection,
+                                                  schema=schema, table_name=table_name, **kw)
 
-        if len(tables) > 0:
+        if len(tables) == 1:
             # get_columns_sql originally returned all columns of all tables if table_name is None,
             # we follow this behavior here for compatibility. However, the documentation for Dialects
             # does not mentions this behavior:
             # https://docs.sqlalchemy.org/en/13/core/internals.html#sqlalchemy.engine.interfaces.Dialect
-            quoted_schema_string = self.quote_string_value(tables[0][1])
+            quoted_schema_string = self.quote_string_value(tables[0].table_schem)
+            quoted_table_string = self.quote_string_value(tables[0].table_name)
             sql_stmnt = \
-                "/*snapshot execution*/ SELECT " \
-                "column_name, column_type, column_maxsize, column_num_prec, column_num_scale, " \
-                "column_is_nullable, column_default, column_identity, column_is_distribution_key, column_table " \
-                "FROM sys.exa_all_columns WHERE column_schema = {schema} " \
-                "ORDER BY column_ordinal_position" \
-                    .format(schema=quoted_schema_string)
+                "/*snapshot execution*/ " + \
+                self.get_column_sql_query_str() \
+                    .format(schema=quoted_schema_string, table=quoted_table_string)
             rp = list(connection.execute(sql.text(sql_stmnt)))
-
-            return rp
+            return list(rp)
```

Comment:
Why was `return rp` replaced by `return list(rp)` if the previous line is already casting to list?

Reply:
Likely, a leftover of some debugging, I am going to change it back.